client-java/src/main/java/uk/gov/gchq/palisade/client/util/Configuration.java

```diff
@@ -0,0 +1,366 @@
+     */
+    public static final String KEY_STATE_PATH = "state.path";
+
+    /**
+     * palisade service scheme
+     */
+    public static final String KEY_SERVICE_PS_SCHEME = "service.palisade.scheme";
+
+    /**
+     * palisade service port
+     */
+    public static final String KEY_SERVICE_PS_PORT = "service.palisade.port";
+
+    /**
+     * palisade service context
+     */
+    public static final String KEY_SERVICE_PS_CONTEXT = "service.palisade.context";
+
+    /**
+     * palisade service endpoint name
+     */
+    public static final String KEY_SERVICE_PS_ENDPOINT = "service.palisade.endpoint";
+
+    /**
+     * filtered resource service scheme
+     */
+    public static final String KEY_SERVICE_FRS_SCHEME = "service.filteredResource.scheme";
+
+    /**
+     * filtered resource service port
+     */
+    public static final String KEY_SERVICE_FRS_PORT = "service.filteredResource.port";
+
+    /**
+     * filtered resource service context
+     */
+    public static final String KEY_SERVICE_FRS_CONTEXT = "service.filteredResource.context";
+
+    /**
+     * filtered resource service endpoint
+     */
+    public static final String KEY_SERVICE_FRS_ENDPOINT = "service.filteredResource.endpoint";
+
+    /**
+     * data service context
+     */
+    public static final String KEY_SERVICE_DS_CONTEXT = "service.data.context";
+
+    /**
+     * data service endpoint
+     */
+    public static final String KEY_SERVICE_DS_ENDPOINT = "service.data.endpoint";
+
+    /**
+     * number of download threads
+     */
+    public static final String KEY_DOWNLOAD_THREADS = "download.threads";
+
+    /**
+     * path for receiver to place files
+     */
+    public static final String KEY_RECEIVER_FILE_PATH = "receiver.file.path";
+
+    /**
+     * the class of the receiver
+     */
+    public static final String KEY_RECEIVER_FILE_CLASS = "receiver.file.class";
+
+    /**
+     * receiver file template
+     */
+    public static final String KEY_RECEIVER_FILE_TEMPLATE = "receiver.file.template";
+
+    private final Map<String, Object> properties;
+
+    private Configuration(final Map<String, Object> properties) {
+        this.properties = properties;
+    }
+
+    /**
+     * Creates a new configuration instance with the default overridden by values
+     * from the provided map
+     *
+     * @param properties The property overrides
+     * @return a new configuration instance with the default overridden by values
+     *         from the provided map
+     */
+    public static Configuration from(final Map<String, Object> properties) {
+        return fromDefaults().merge(properties);
+    }
+
+    /**
+     * Returns a new configuration instance loaded with all defaults
+     *
+     * @return a new configuration instance loaded with all defaults
+     */
+    public static Configuration fromDefaults() {
+
+        try {
+
+            var url = Thread.currentThread().getContextClassLoader().getResource("palisade-client.yaml");
```

Comment:
There is no Spring in the project. There are a couple of configuration libraries I was looking at that would be good candidates, but went with the simplest solution to avoid dependency creep.